server/routes/monitor.ts

```diff
@@ -39,70 +39,160 @@ export default function routes({ serviceCatalogueService, redisService, dataFilt
     const monitorName = getMonitorName(req)
     logger.info(`Request for /monitor/${monitorType}/${monitorName}`)
 
+    // If we have a product name, look up its ID
+    let monitorId = 0
+    if (monitorType === 'product' && monitorName) {
+      try {
+        const products = await serviceCatalogueService.getProducts({})
+        logger.info(`Looking for product with name that matches: ${monitorName}`)
+
+        // Try to match by name, slug, or formatted name
+        const matchingProduct = products.find(p => {
+          // Try exact name match
+          if (p.attributes.name === monitorName) return true
```

Comment:
Also ideally linting would pick up this, we'd normally not have single line `if`